src/ucp/rndv/proto_rndv.c

```diff
@@ -415,18 +415,32 @@ void ucp_proto_rndv_ctrl_probe(const ucp_proto_rndv_ctrl_init_params_t *params,
     remote_proto_init = &remote_proto_select->proto_init;
 
     /* Add variants for each remote proto */
+    ucs_trace("add variants for %s: min_length=%zu max_length=%zu",
+              ucp_proto_id_field(init_params->proto_id, name),
+              params->super.min_length, params->super.max_length);
+    ucs_log_indent(+1);
     ucs_array_for_each(remote_proto, &remote_proto_init->protocols) {
-        remote_caps = &remote_proto->caps;
-        /* Skip empty variants and reconfig proto */
-        if ((remote_caps->num_ranges == 0) || (remote_proto->proto_id == 0)) {
+        /* Skip reconfig proto */
+        if (remote_proto->proto_id == 0) {
```

Comment:
can it be a better check than comparing proto_id to 0? for example, UCP_PROTO_FLAG_INVALID?